src/invoices/InvoiceDetails/BatchGroupValue/BatchGroupValue.js

```diff
@@ -0,0 +1,46 @@
+import React, {
+  useState,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+
+import { stripesConnect } from '@folio/stripes/core';
+import { KeyValue } from '@folio/stripes/components';
+
+import { batchGroupByPropResource } from '../../../common/resources';
+
+const BatchGroupValue = ({ id, label, mutator }) => {
+  const [batchGroup, setBatchGroup] = useState();
+
+  useEffect(
+    () => {
+      setBatchGroup();
+
+      if (id) {
+        mutator.invoiceBatchGroup.GET()
+          .then(setBatchGroup);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [id],
+  );
+
+  return (
+    <KeyValue
+      label={label}
+      value={batchGroup?.name}
```

Comment:
IMHO, we should check if batchGroup and show spinner (Loading). Also this requires to add .catch for API request and set {} there

Reply:
this should be changed